nyxtools/vector.py

```diff
@@ -0,0 +1,253 @@
+    shutter_lag_dist = FCpt(EpicsSignalRO, "{prefix}Val:{motor_name}ShutLagDist-I")
+
+    def __init__(self, prefix, motor_name=None, **kwargs):
+        self.motor_name = motor_name
+        super().__init__(prefix, **kwargs)
+
+
+class VectorProgram(Device):
+    """
+    Wraps PVs that control the vector program.
+    """
+
+    #
+    # Configuration
+    #
+
+    # Exposure per sample (ms)
+    exposure = Cpt(VectorSignalWithRBV, "Val:Exposure")
+
+    # Number of samples
+    num_samples = Cpt(VectorSignalWithRBV, "Val:NumSamples")
+
+    # "Buffer" motion time (ms)
+    buffer_time = Cpt(VectorSignalWithRBV, "Val:BufferTime")
+
+    # Shutter opening / closing time (ms)
+    shutter_time = Cpt(VectorSignalWithRBV, "Val:ShutTime")
+
+    # Shutter lag (ms)
+    shutter_lag_time = Cpt(VectorSignalWithRBV, "Val:ShutLagTime")
+
+    # Whether to expose (whether to open the shutter during acquisition)
+    expose = Cpt(EpicsSignal, "Expose-Sel")
+
+    # Whether to actually execute the motion
+    # If calc_only==True, no motion is performed but the vector profile is calculated
+    # (useful for debugging)
+    calc_only = Cpt(EpicsSignal, "CalcOnly-Sel")
+
+    # Whether to pause the vector after backing up but before the start of the motion
+    hold = Cpt(EpicsSignal, "Hold-Sel")
+
+    #
+    # Individual motor configuration / status
+    #
+
+    # Omega motor
+    o = Cpt(VectorMotor, "O")
+
+    # X motor
+    x = Cpt(VectorMotor, "X")
+
+    # Y motor
+    y = Cpt(VectorMotor, "Y")
+
+    # Z motor
+    z = Cpt(VectorMotor, "Z")
+
+    #
+    # Debugging
+    #
+
+    # Calculated data acquisition duration (ms)
+    data_acq_duration = Cpt(EpicsSignalRO, "Val:DataAcqDuration-I")
+
+    # Calculated maximum time to reach desired speeds (ms)
+    max_time_to_speed = Cpt(EpicsSignalRO, "Val:MaxTimeToSpeed-I")
+
+    #
+    # Status
+    #
+
+    # Indicates whether a vector motion is running
+    running = Cpt(EpicsSignalRO, "Sts:Running-Sts")
+
+    # Current state of the vector:
+    #   Idle, Backup, Holding or Acquiring
+    state = Cpt(EpicsSignalRO, "Sts:State-Sts")
+
+    # Error reported by the vector program:
+    #   None, Aborted, Zero Exposure, Too Fast, Zero Shutter, Too Slow
+    error = Cpt(EpicsSignalRO, "Sts:Error-Sts")
+
+    #
+    # Commands
+    #
+
+    # Start a vector motion
+    go = Cpt(EpicsSignal, "Cmd:Go-Cmd")
+
+    # Proceed with the vector motion when it is paused
+    # Only needed if hold==True
+    proceed = Cpt(EpicsSignal, "Cmd:Proceed-Cmd")
+
+    # Abort the current vector motion
+    abort = Cpt(EpicsSignal, "Cmd:Abort-Cmd")
+
+    # Set all vector motors start and end position to their current RBV values
+    sync = Cpt(EpicsSignal, "Cmd:Sync-Cmd")
+
+    def run(
+        self,
+        o: Tuple[float,float], x: Tuple[float,float], y: Tuple[float, float], z: Tuple[float,float],
+        exposure: float, num_samples: float, buffer_time: float, shutter_lag_time: float, shutter_time: float):
```

Comment:
Should we comment somewhere that the times here are in ms?

Reply:
I added a `_ms` suffix to the parameters